src/openfhe.jl

```diff
@@ -74,28 +70,124 @@ end
 
 """
     init_rotation!(context::SecureContext{<:OpenFHEBackend}, private_key::PrivateKey,
-                   shifts)
+                   shape::NTuple{N, Integer}, rotation_index...)
 
-Generate rotation keys for use with `OpenFHE.EvalRotate` using the `private_key` and for the
-rotation indexes in `shifts`. The keys are stored in the given `context`.
-Positive shift defines rotation to the right, e.g. a rotation with a shift 1:
-[1, 2, 3, 4] -> [4, 1, 2, 3].
-Negative shift defines rotation to the left, e.g. a rotation with a shift -1:
-[1, 2, 3, 4] -> [2, 3, 4, 1].
-
-Note: Here, indexes stored in `shifts` have reversed sign compared to rotation indexes used in
-OpenFHE.
+Generate all required rotation keys for use with `circshift` for the
+rotation indexes in `rotation_index` using the `private_key`. The keys are stored in the given `context`.
 
 See also: [`SecureContext`](@ref), [`OpenFHEBackend`](@ref), [`PrivateKey`](@ref)
 """
 function init_rotation!(context::SecureContext{<:OpenFHEBackend}, private_key::PrivateKey,
-                        shifts)
+                        shape::NTuple{N, Integer}, rotation_index...) where N
     cc = get_crypto_context(context)
-    OpenFHE.EvalRotateKeyGen(cc, private_key.private_key, -shifts)
+    # Get indexes for precompilation
+    indexes = get_index(context, shape, rotation_index)
```

Comment:
```suggestion
    indices = get_index(context, shape, rotation_index)
```
`indices` is the correct plural of `index`